src/sql/workbench/parts/objectExplorer/browser/treeSelectionHandler.ts

```diff
@@ -57,12 +60,18 @@ export class TreeSelectionHandler {
 		let specificSelection = selection[0];
 
 		if (this.isMouseEvent(event)) {
-			if (this._lastClicked === specificSelection) {
+			if (this._lastClicked !== undefined) {
 				clearTimeout(this._clickTimer);
-				sendSelectionEvent(event, selection, true);
-				return;
+				let lastSpecificClick = this._lastClicked[0];
+
+				if (lastSpecificClick === specificSelection) {
+					sendSelectionEvent(event, selection, true);
+					return;
+				} else {
+					sendSelectionEvent(event, this._lastClicked, false);
+				}
 			}
-			this._lastClicked = specificSelection;
+			this._lastClicked = selection;
 		}
 
 		this._clickTimer = setTimeout(() => {
```

Comment:
vscode also has a class for wrapping timeouts, https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/common/async.ts#L496 if you want to use it.

Reply:
Oo this looks good!